### ISSUE TITLE
use Poetry with pyproject.toml

### ISSUE THREAD

**Issue by veghdev** [vizzuhq/ipyvizzu#228]
Instead of setup.py create a pyproject.toml file.

- [x] add a pyproject.toml draft file
- [x] separate extras dependencies (fugue) in the pyproject.toml file
- [ ] modify github workflows
- [ ] use task runner (for example poe) instead of makefile #197

**Comment by veghdev** [comment]
Hi @jharibo!
Thank you for your help.

We would like to use poetry with pyproject.toml, instead of setup.py. But we will switch only after all related tasks are completed.
So in ipyvizzu there is a [draft-pyproject.toml](https://github.com/vizzuhq/ipyvizzu/blob/main/draft-pyproject.toml) file, and it will be good to make a similar in ipyvizzu-story.

The first task is in the draft file, fugue is under 'dependencies ', but its only an extra.

The other part is more complex, we would like to get rid of the makefile, and use a task runner which fits better in a python environment and it would also be good, if it only depends on python and not on os. #197 #229 
Now, we compile dev dependecies with pip-compile, I think this should also be changed if we want to support multiple python versions and os.

If you start working with a specific part-task, please comment under the related issue :) and we can separate the task list into more sub-tasks in the descriptions.

Feel free to submit a separate PR for each subtask you're working on. Thanks.
